src/vs/workbench/contrib/debug/browser/debugHover.ts

```diff
@@ -176,7 +182,52 @@ export class DebugHoverWidget implements IContentWidget {
 			options: DebugHoverWidget._HOVER_HIGHLIGHT_DECORATION_OPTIONS
 		}]);
 
-		return this.doShow(pos, expression, focus);
+		return expression;
+	}
+
+	private async showSelectionEvaluation(selection: Selection, model: ITextModel, session: IDebugSession): Promise<IExpression | undefined> {
+		if (!session.capabilities.supportsEvaluateForHovers) {
+			return undefined;
+		}
+
+		const selectedText = model.getValueInRange(selection);
+		const expression = new Expression(selectedText);
+		await expression.evaluate(session, this.debugService.getViewModel().focusedStackFrame, 'hover');
+
+		if (!expression || (expression instanceof Expression && !expression.available)) {
+			return undefined;
+		}
+
+		return expression;
+	}
+
+	async showAt(range: Range, focus: boolean): Promise<void> {
```

Comment:
This looks good, however can we rewrite this such that we introduce minimal changes. 
There is a changing of methods.
What we basically need is same as before, read the setting, read the selection. If setting true and there is a useful selection do evaluation, otherwise as before. No changing of methods is needed, we just maybe need one new method.

Reply:
I think these refactorings are needed to not cause code duplications. However I accepted you other reviews, I think you should reconsider your last suggestion.
Thank you.